### PR TITLE
Initial cloneset controller

### PR DESCRIPTION
**Pull request by FillZpp**
### Ⅰ. Describe what this PR does
Initial cloneset controller, does not contain management of pods create/delete/update yet.

### CODE REVIEW

File: pkg/controller/cloneset/cloneset_controller.go [hunk 1]
```diff
@@ -87,18 +131,214 @@ type ReconcileCloneSet struct {
...
+func (r *ReconcileCloneSet) truncateHistory(
+	cs *appsv1alpha1.CloneSet,
+	pods []*v1.Pod,
+	revisions []*apps.ControllerRevision,
+	current *apps.ControllerRevision,
+	update *apps.ControllerRevision,
+) error {
+	historyRevisions := make([]*apps.ControllerRevision, 0, len(revisions))
+	// mark all live revisions
+	live := map[string]bool{current.Name: true, update.Name: true}
```
Comment (furykerry) [inline pkg/controller/cloneset/cloneset_controller.go]: 用下这个？ k8s.io/apimachinery/pkg/util/sets

File: pkg/controller/cloneset/utils/cloneset_utils.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	kubecontroller "k8s.io/kubernetes/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ControllerKind = appsv1alpha1.SchemeGroupVersion.WithKind("CloneSet")
+
+func GetActivePods(reader client.Reader, namespace string) ([]*v1.Pod, error) {
```
Comment (furykerry) [inline pkg/controller/cloneset/utils/cloneset_utils.go]: exported functions or variables should be commented

File: pkg/controller/cloneset/cloneset_controller.go (hunk 1 above)
Comment (FillZpp) [reply pkg/controller/cloneset/cloneset_controller.go]: done

File: pkg/controller/cloneset/utils/cloneset_utils.go (hunk 2 above)
Comment (FillZpp) [reply pkg/controller/cloneset/utils/cloneset_utils.go]: done

**Pull request merged**